src/meta/src/stream/stream_manager.rs

```diff
@@ -149,26 +149,34 @@ impl CreatingStreamingJobInfo {
     async fn cancel_jobs(
         &self,
         job_ids: Vec<JobId>,
-    ) -> (HashMap<JobId, oneshot::Receiver<bool>>, Vec<JobId>) {
+    ) -> MetaResult<(HashMap<JobId, oneshot::Receiver<bool>>, Vec<JobId>)> {
         let mut jobs = self.streaming_jobs.lock().await;
         let mut receivers = HashMap::new();
         let mut background_job_ids = vec![];
         for job_id in job_ids {
             if let Some(job) = jobs.get_mut(&job_id) {
                 if let Some(shutdown_tx) = job.shutdown_tx.take() {
                     let (tx, rx) = oneshot::channel();
-                    if shutdown_tx.send(tx).is_ok() {
-                        receivers.insert(job_id, rx);
-                    } else {
-                        tracing::warn!(id=?job_id, "failed to send canceling state");
+                    match shutdown_tx.send(tx) {
+                        Ok(()) => {
+                            receivers.insert(job_id, rx);
+                        }
+                        Err(_) => {
+                            return Err(anyhow::anyhow!(
+                                "failed to send shutdown signal for streaming job {}: receiver dropped",
```

Comment:
Why would the receiver for the streaming job be dropped? Is it due to concurrent cancellations?